lib/repo_tools.pm

```diff
@@ -558,4 +559,20 @@ sub verify_software {
     return '';
 }
 
+=head2 ensure_ca_certificates_suse_installed
+    ensure_ca_certificates_suse_installed();
+
+This functions checks if ca-certificates-suse is installed and if it is not it adds the repository and installs it.
+
+=cut
+
+sub ensure_ca_certificates_suse_installed {
+    return unless (is_sle || is_tumbleweed || is_leap);
```

Comment:
well with such condition there is no way this return will be called ever :D 
which means you can just drop it

Reply:
There is SLE micro which fails `is_sle` IIRC

Reply:
ok agree . but I generally don't see any good reason why we need to limit someone to call this function . Do you see something ?

Reply:
Not really. Especially the `ensure` part is wrong with this condition, as on SLE Micro it won't ensure anything

Reply:
Ok, then I will just drop that line. And if `generate_version()` retrieves something wrong, we will fail on `zypper ar` which is fine for me!